staff/asier/react-spotify-api-promises/scripts/main.jsx

```diff
@@ -0,0 +1,312 @@
+'use strict'
+
+// https://stackoverflow.com/questions/24502898/show-or-hide-element
+
+
+class SpotyApp extends React.Component {
+    constructor() {
+        super()
+        this.state = {
+            artists: [],
+            showArtists: true,
+            albums: [],
+            showAlbums: false,
+            songs: [],
+            showSongs: false,
+            songId: []
+        }
+    }
+
+    // check the results and look up for deafult pictures. 
+
+    fixResultsWithoutPictures(sourceData) {
+        for (let i = 0; i < sourceData.length; i++) {
+            for (let prop in sourceData[i]) {
+                if (sourceData[i].images.length === 0) {
+                    // si el item no tiene imágenes, le establezco una por defecto 
+                    sourceData[i].images.push({ height: 600, url: "img/noImage.jpg", width: 600 });
+                }
+            }
+        }
+        return sourceData;
+    }
+
+    // click on "back to artists" button
+
+    goBack = () => {
+        this.setState({ showArtists: true })
+        this.setState({ showAlbums: false })
+        this.setState({ showSongs: false })
+
+    }
+
+
+    fetchArtists = (query) => {
+
+        this.setState({ showArtists: true })
+        this.setState({ showAlbums: false })
+        this.setState({ showSongs: false })
+
+        spotifyApi.searchArtists(query)
+            .then((listOfArtists) => {
+                let images = this.fixResultsWithoutPictures(listOfArtists)
+                this.setState({
+                    artists: images
+                })
+            })
+            .catch(err => console.error("err -->", err))

```

Comment:
MAY error catching be as simple as just passing the console.error function as handler to just directly mark error in console, without the need of an extra wrapping arrow function:

.catch(console.error)

SAME for other analog cases below...